hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MultiVersionConcurrencyControl.java

```diff
@@ -272,33 +252,59 @@ public long getWritePoint() {
    */
   @InterfaceAudience.Private
   public static class WriteEntry {
-    private final long writeNumber;
+    private long writeNumber = 0L;
     private boolean completed = false;
+    private static final Logger LOG = LoggerFactory.getLogger(WriteEntry.class);
 
-    WriteEntry(long writeNumber) {
-      this.writeNumber = writeNumber;
+    synchronized WriteEntry markCompleted(boolean completed) {
```

Comment:
Do we need to pass a boolean here? We are always setting complete to true? We ever set it to false? This  methods should take no parameters?

Reply:
when reuse the thread local `WriteEntry`, we must set it incomplete